[WIP] Overthrow gamemode

Carbonhell opened the pull request:
:cl: 
add: A new gamemode called overthrow has been added. A few syndicate mutineers spawn at roundstart, with special items such as a converting implanter and a hacked AI board, their objectives are to stage a bloodless coup. Each initial agent has its own syndicate team, and said teams can choose to cooperate or not. The team with the most points, given by the objectives, wins at roundend.
/:cl:

[why]: This is for Naksu's gamemode bounty, the guidelines are mainly his, basically the objective of those dudes is to convert/exile/kill heads and an eventual target, along with taking control of the AI. Each objective gives a certain amount of points, you get more by converting a head to your team (1.5x multiplier), if a head is converted by another team, or exiled somehow, the multiplier is 1x. If the antags are boring fucks and murderbone, the multiplier is 0.5x, to encourage conversion and exiling without necessarily avoiding killing. Each head has a base point value defined in the overthrow objective dm with defines which i will paste here too:
#define CAPPTS		5
#define AIPTS		5
#define HOPPTS		4
#define HOSPTS		4
#define TARGETPTS	4
#define CEPTS		3
#define CMOPTS		3
#define RDPTS		3
#define WARDENPTS	2
#define SECPTS		1

multipliers:
#define CONVERTED_OURS	1.5
#define CONVERTED		1
#define EXILED			1
#define KILLED			0.5

there are defines for warden and sec too, i am not totally sure whether to make converting them give points too or not, but right now it's not coded in.
Leaders can convert others (aka wake them, the "lore" is about sleeping syndicate agents") which will "reveal" their antag status, along with giving them a storage implant with a random item (mostly syndicate items that cost < 5 TCs, hand picked into a static list of the antagonist datum)
Round ends once shuttle reaches centcom, points are calculated and the team with the most points is declared winner. The crew has to fight /all/ teams to win, obviously. Teams can also fight each other to get more points, sabotage others to make them get less points, and such.
I did some very basic tests(basically checking if the gamemode itself runs without runtimes), it sorta works now but it needs futher fixes.

Carbonhell committed:
Initial draft of the overthrow gamemode

Carbonhell committed:
Initial draft of the overthrow gamemode

Carbonhell committed:
Initial draft of the overthrow gamemode

Carbonhell committed:
Initial draft of the overthrow gamemode

81Denton committed:
Adds more wiki books/manuals

81Denton committed:
renamed atmos manual, added toxins manual suicide

81Denton committed:
[DNM] Adds more destination tagger locations

81Denton committed:
Merge remote-tracking branch 'tgstation/master' into disposal_destinations_v2

81Denton committed:
Properly fixes Delta mail sorting
dv

81Denton committed:
adds sortType commentary

81Denton committed:
Adds det office junctions

81Denton committed:
hhhh

SpaceManiac committed:
Fix freezers which start on appearing on when off

ninjanomnom committed:
Handle shuttle registration from inside the template

AnturK committed:
Fixes syringe injection logging for non-mobs.

AnturK committed:
Fixes flash logging for emp/remote activation.

AnturK committed:
Minor puzzle performance improvement

Carbonhell committed:
Fixes equipping procs and an objective runtime

Carbonhell committed:
Fixes equipping procs and an objective runtime

Carbonhell committed:
Fixes equipping procs and an objective runtime

Carbonhell committed:
Fixes equipping procs and an objective runtime

ExcessiveUseOfCobblestone commented:
It's a better rev without updating rev smh

Tlaltecuhtli commented:
rev but you kill ppl with ai?

XDTM commented:
Wait, so every single crewmember is a sleeper agent?

Carbonhell committed:
Merge branch 'tgcode/overthrow' of https://github.com/Carbonhell/HippieStation into tgcode/overthrow

Carbonhell commented:
That's just the fluff behind it, to explain them magically obtaining a storage implant with an item in it, but it's just like rev conversion @XDTM

XDTM commented:
Seems like there's an incentive for each team to kill the opposing team's convertees, at that will negate the point disparity (0.5 for both instead of 1.5 for one and 1.0 for the other), unless i misunderstoon point attribution.

XDTM commented:
I'd go as far as to suggest an OOC directive for the teams to keep the crew alive (self-defense aside), because i feel like the playerbase really has an issue of "antag=can murderbone" in most cases

Carbonhell commented:
Technically yes, that's where the team is supposed to do some teamwork and defend the targets they convert instead of going a la rev aka convert and forget
also i'd rather fix any issue codewise since i'm mainly a hippie coder and i want this to work on all forks aswell without rules getting in

Carbonhell committed:
Fixes logging compile error

Carbonhell committed:
Fixes logging compile error

ExcessiveUseOfCobblestone commented:
How do you lose?

Mickyan commented:
Interesting idea, I have mixed feelings about giving everyone random traitor items, and the "lore" behind it is rather flimsy, on a minor note. 

What if instead each leader had an uplink that generated TC through conversions/gaining points (may start empty or with only a small amount of TC). Gives the leader more power to set a clear strategy and distribute equipment.

There's the added benefit that people can no longer metagame the round type with 100% certainty just because they find a traitor item laying around

Carbonhell commented:
the "just because they find a traitor item laying around" isn't really metagame-able, or more than it is currently.  Also, points can change in the round, if you convert a head the modifier is 1.5x but another team can kill said head to lower it to 0.5x, so i assume you mean something like checking every x minutes the current amount of points and generating y telecrystals based on said amount, which could be done but could also make those antags really powerful and give them too much traitor gear, when simply having a head on your side helps your team a lot

Carbonhell commented:
Only one team can win at round end, aka the one with the most points. You lose if your team gets fully killed,or another team gathers more points than you

fludd12 commented:
I dunno, I find the whole Paranoia tabletop rpg style, everyones-a-sleeper-agent-of someone-thing to be pretty funny. Might need some more development to work out, but I like it

Mickyan commented:
I didn't mean to say this PR would cause metagaming the round

I think periodically checking for points to award tc could make for some interesting tactics, do you rush your objectives and risk getting found out to get some good equipment early on or play it safe without the help of traitor items?
It's just a matter of balancing the TC intake to be sensibly low. Give them a max cap if necessary.

SpaceManiac committed:
Clean up bar sign code (#39787)

Iamgoofball commented:
>>>conversion gamemodes

ExcessiveUseOfCobblestone commented:
If only one team can win why would they help each other?

If someone is converted and I can't convert them again OR the switch conversion process is too long i'm just going to off them. I'll very readily take the 50% vs risking the other team letting them escape and get 150% percent.

One thing I can't wait to see is the Policy Discussion, or even better the first ahelp, regarding "is killing a person worth 150 for any reason detrimental to the team?"

81Denton committed:
hmm

81Denton committed:
Merge remote-tracking branch 'tgstation/master' into disposal_destinations_v3

Carbonhell committed:
Adds some basic admin commands

Carbonhell committed:
Adds some basic admin commands

81Denton committed:
I love me some map merge conflicts

octareenroon91 commented:
Is this likely to play out like gangs?

Anonmare committed:
Fix multitools hitting SSUs (#39780)

Nanotrasen reminds their employees that their tools have feelings too.

Cyberboss committed:
Merge pull request #39758 from 81Denton/disposal_destinations_v3

Adds more destination tagger locations (now with mapdiffbot)

tgstation-server committed:
Automatic changelog generation for PR #39758 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

SpaceManiac committed:
Slash SSatmos init time by 93%

tgstation-server committed:
Automatic changelog compile, [ci skip]

Commit:
Fix the hunger alert appearing forever (#39807)

fuk

tgstation-server committed:
Automatic changelog generation for PR #39807 [ci skip]

silicons committed:
Fix experimental cloners mixing up ckeys and names (#39804)

Experimental cloners will no longer print clones with people's ckeys instead of
names when interfaced by a regular cloning console.

Argument order was missing an argument.

intrnlerr committed:
Fix tank temperature check checking pressure instead (#39805)

Makes temperature actually temperature.

tgstation-server committed:
Automatic changelog generation for PR #39805 [ci skip]

81Denton committed:
Add ore silo examine message, add it to runtimestation (#39792)

I added an examine message to the ore silo:

    [src] can be linked to techfabs, circuit printers and protolathes with a
    multitool.

Also added it to runtimestation, which required me to expand the area a little.

81Denton committed:
Fixes Meta disposals junction

Cyberboss committed:
Merge pull request #39826 from AutomaticFrenzy/patch/atmos-init

Slash SSatmos init time by 93%

tgstation-server committed:
Automatic changelog generation for PR #39826 [ci skip]

Cyberboss committed:
Merge pull request #39812 from AnturK/puzzleperf

Minor puzzle performance improvement

Cyberboss committed:
Merge pull request #39803 from AutomaticFrenzy/patch/freezers

Fix freezers which start on appearing on when off

tgstation-server committed:
Automatic changelog generation for PR #39803 [ci skip]

Cyberboss committed:
Removes security mode checks (#39790)

Cyberboss committed:
Merge pull request #39806 from ninjanomnom/template-registration

Handle shuttle registration from inside the template

tgstation-server committed:
Automatic changelog generation for PR #39806 [ci skip]

Cyberboss committed:
Merge pull request #39809 from AnturK/syringeinjectlogfix

Fixes syringe injection logging for non-mobs.

Cyberboss committed:
Merge pull request #39810 from AnturK/flashwithoutuser

Fixes flash logging for emp/remote activation.

Nabski89 commented:
The big changes from original gangs is that there's not a dominator/hard end the round object, and there's an attempt to discourage mass murder by a point scheme.

Giving someone recurring points based off thing makes it very gangy where if you don't have your boss alive you are fucked, or even worse robustin gangs where everyone just is constantly shitting out guns and knives.

ExcessiveUseOfCobblestone commented:
If it doesn't play like gang it will play like sling where the important people hide in [location] while the minions go out and kill everyone else.

vuonojenmustaturska committed:
henk (#39841)

optimumtact committed:
Merge pull request #39837 from 81Denton/god-dangit-bobby

Fixes Meta disposals junction

octareenroon91 commented:
>discourage mass murder by a point scheme

Would it work even if a live neutral player was worth more than a dead foe?

optimumtact committed:
Make the inventory helper overlay less impactful (#39824)

Reduces alpha and has a traffic light system for the items

-> red ghost when the item can't be placed
-> green ghost when it can
-> full solid colour item, object is in the inventory

This is so you can differentiate the ghost view from the actual object entering
the slot, which was sticky's primary complaint

I moved the cut of overlays only to mouse out, since if that somehow isn't
called we have bigger problems than overlays doubling up. INB4 this causes
server crashes.

Side bonus it removes one image operation, it's worth investigating a cache for
this as well I would say, a simple rotating buffer list of the last x many
images.

Tested and ready to go

tgstation-server committed:
Automatic changelog generation for PR #39824 [ci skip]

SpaceManiac committed:
Copy spritesheet files to the cache rather than persisting them (#39802)

AnturK commented on code/game/gamemodes/overthrow/overthrow.dm:
```diff
@@ -0,0 +1,46 @@
+		var/datum/mind/sleeping_agent = antag_pick(antag_candidates)
+		antag_candidates -= sleeping_agent
+		initial_agents += sleeping_agent
+		sleeping_agent.restricted_roles = restricted_jobs
```
You're not setting special role anywhere so it's possible for them to be kicked back to lobby and other unpleasantness.

Carbonhell replied:
that's a thing?what the fuck, why does the special_role var still exist
anyway,fixing

AnturK commented on code/modules/antagonists/overthrow/overthrow.dm:
```diff
@@ -0,0 +1,128 @@
+									/obj/item/soap/syndie, /obj/item/card/id/syndicate, /obj/item/storage/box/syndie_kit/chameleon)
+
+/datum/antagonist/overthrow/on_gain()
+	create_team()
```
You don't need to call it manually, `create_team` is called in `add_antag_datum`, you should be passing the already existing team there. Personally i feel like splitting naming and team creation is good idea here.

AnturK commented on code/modules/antagonists/overthrow/overthrow.dm:
```diff
@@ -0,0 +1,128 @@
+	.["Give overthrow boss equip"] = CALLBACK(src,.proc/equip_overthrow_boss)
+
+/datum/antagonist/overthrow/proc/update_overthrow_icons_added(datum/mind/traitor_mind)
+	var/datum/atom_hud/antag/overthrowhud = GLOB.huds[team.hud_entry_num]
```
You should probably remove these at some point if team gets deleted.

AnturK commented on code/modules/antagonists/overthrow/overthrow.dm:
```diff
@@ -0,0 +1,128 @@
+	to_chat(owner.current, "<B><font size=3 color=red>You are a syndicate sleeping agent. Your job is to stage a swift, fairly bloodless coup. </font></B>")
+
+/datum/antagonist/overthrow/boss
+	name = "Syndicate initial eutineers"
```
typo

AnturK commented on code/modules/antagonists/overthrow/overthrow_team.dm:
```diff
@@ -0,0 +1,19 @@
+	heads.find_target()
+	objectives += heads
+	// AI objective
+	var/datum/objective/overthrow_AI/AI = new()
```
Didn't set owner here.

AnturK commented on code/game/gamemodes/overthrow/objective.dm:
```diff
@@ -0,0 +1,121 @@
+	var/base_points = 0
+	for(var/i in targets)
+		var/datum/mind/M = i
+		if(M && M.current)
```
Blown up don't count ?

WJohn committed:
Removes now obsolete floorgrime path and icon (#39816)

* Removes now obsolete floorgrime path and icon

* Whoops.

* Fixes me being a dummy.

Jordie0608 committed:
fix rankchange button in player panel (#39834)

tgstation-server committed:
Automatic changelog compile, [ci skip]

BlueNothing committed:
Allows video camera circuits to be seen through networks aside from the science network. (#39657)

Also allows camera bugs to see through borg cameras and camera circuits, and alphabetizes the camera bug camera list.

tgstation-server committed:
Automatic changelog generation for PR #39657 [ci skip]

PKPenguin321 committed:
IC GPS Update (#39630)

GPS circuit now has a 4th output, placing X,Y,Z all in a string.
2 new converters: Rel to Abs, and Adv Rel to Abs.
Rel to Abs takes a set of relative and a set of absolute coordinates, and converts the relative one to absolute. 1 complexity.
Adv Rel to Abs takes a set of relative coordinates and converts it to absolute without the need for an already known set of absolute coordinates. 2 complexity.

tgstation-server committed:
Automatic changelog generation for PR #39630 [ci skip]

MrsTonedOne committed:
/mob/dead no longer has a throwforce

Cyberboss committed:
Merge pull request #39859 from tgstation/MrStonedOne-patch-2

/mob/dead no longer has a throwforce

drderp3635 commented:
there are two suggestions i have for this.
1-give some kind of point value to regular crew members, a half point would probably do
2-give a war-type option to the teams, the leaders would have some form of voting for this, and if all leaders unanamously agree to it, it will make an anouncement to the station and the teams will have to surpass some number of points which requires nearly everyone to be converted.

tgstation-server committed:
Automatic changelog compile, [ci skip]

SpaceManiac committed:
Fix various things not returning an initialize hint (#39843)

* Fix RCDs not returning an initialize hint

* Add Initialize hints to cult blade and telecomms allinone

Commit:
isnum() checks for squeak component initialisation (#39847)

if you supply step_delay_override or use_delay_override as 0 then component
just ignores these new values despite them being valid options

by using isnum() checks, these values can actually be used for this

Cyberboss committed:
Fix getrev runtime for the billionth time (#39845)

tgstation-server committed:
Automatic changelog compile, [ci skip]

ninjanomnom committed:
Make radiation waves return QDEL_HINT_IWILLGC (#39851)

None of their functionality depends on being referenced by anything so the only
possible issue is if some future coder fucks it up

Carbonhell committed:
special_role, hud deletion, typofix, owner set and blownup checks

Carbonhell committed:
special_role, hud deletion, typofix, owner set and blownup checks

Carbonhell commented:
Changes suggested by Antur applied, but i still need to properly test this.

Carbonhell commented on code/game/gamemodes/overthrow/overthrow.dm:
```diff
@@ -0,0 +1,47 @@
+	config_tag = "overthrow"
+	antag_flag = ROLE_OVERTHROW
+	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer")
+	required_players = 1 //20 // the core idea is of a swift, bloodless coup, so it shouldn't be as chaotic as revs.
```
gotta reset those values once the PR is ready

Carbonhell committed:
what the fuck even happened here

Carbonhell commented:
@drderp3635 The syndies already have a benefit from recruiting random people,aka having more people to do their objectives with. I don't think points are necessary, plus you can only convert up to 2-3 guys and you shouldn't be penalized if you cannot find someone to convert imo
the wartype option is completely outside the gamemode's idea of being a bloodless silent coup, i don't think it fits at all honestly

Carbonhell committed:
Fixes implants, roundend log and objectives

Carbonhell commented:
okay i don't know what the hell happened here

Carbonhell committed:
i beg you, unfuck yourself

Carbonhell committed:
Maybe resolves the conflicts...?

Carbonhell commented:
nvm PR is still broken due to 3 unrelated files slipping in,welp

Carbonhell committed:
will this work

Carbonhell committed:
go away useless changes

Carbonhell commented:
I'm gonna reopen this because i have no idea what i did with git/how to fix this
